src/NUnitFramework/framework/Constraints/CollectionOrderedConstraint.cs

```diff
@@ -63,6 +65,10 @@ public CollectionOrderedConstraint Descending
             get
             {
                 descending = true;
+
+                if (this.properties != null && this.properties.Count != 0)
+                    this.properties[this.properties.Keys.Max()] = true;
+
```

Comment:
Yes, 

I initially removed the `Descending` back-end field and did it all with the dictionary. But that broke the test where no sort property was given and it was just testing if a list was sorted descending. 

So I decoupled the functionality and (incorrectly) assumed that if the `Descending` is called first, no sort properties are given later on. 
It should work if it's called in the -`Is.Ordered.By("SomeProp").Descending.ThenBy("AnotherProp").Descending` order. 

I will add another test to handle the current behavior.